extensions/ql-vscode/src/extension.ts

```diff
@@ -457,10 +457,16 @@ async function activateWithInstalledDistribution(
   const localQueryResultsView = new ResultsView(ctx, dbm, cliServer, queryServerLogger, labelProvider);
   ctx.subscriptions.push(localQueryResultsView);
 
-  void logger.log('Initializing variant analysis manager.');
+  void logger.log('Initializing remote queries manager.');
   const rqm = new RemoteQueriesManager(ctx, cliServer, queryStorageDir, logger);
   ctx.subscriptions.push(rqm);
 
+  void logger.log('Initializing variant analysis manager.');
+  const variantAnalysisStorageDir = path.join(ctx.globalStorageUri.fsPath, 'variant-analyses');
+  await fs.ensureDir(variantAnalysisStorageDir);
```

Comment:
Since we're building the storage path off of `ctx`, would we be able to move this logic into a method on the `VariantAnalysisManager` instead of passing in the `storagePath` as a param? 

I realize this `extension.ts` file already does this sort of setup for other commands but it's not very clean or readable. We might also accidentally overwrite some variable here and the extension will just happily go with that.

Reply:
I agree with you that this extension file is already doing too much, but unfortunately moving this logic to the constructor of `VariantAnalysisManager` (or even `VariantAnalysisResultsManager`) is not really possible as-is. In a constructor, it's not possible to use `await`, so we wouldn't be able to call `fs.ensureDir`.

We could also skip the call to `fs.ensureDir` since the `VariantAnalysisResultsManager` should be able to work without the directory existing, but I'm not sure whether there's a reason these calls were added in the first place.

Reply:
That's fair, thanks for looking at it! :)